src/server/scripts/EasternKingdoms/Karazhan/boss_chess_event.cpp

```diff
@@ -0,0 +1,2081 @@
+            {
+                _nextMoveTimer = urand(10 * IN_MILLISECONDS, 20 * IN_MILLISECONDS);
+            }
+
+            if (Creature* medivh = ObjectAccessor::GetCreature(*me, _instance->GetGuidData(DATA_ECHO_OF_MEDIVH)))
+            {
+                if (Creature* target = CAST_AI(npc_echo_of_medivh, medivh->AI())->GetHostileTargetForChangeFacing(me, _currentOrientation))
+                {
+                    CastChangeFacing(me, target);
+                }
+            }
+        }
+    }
+
+    void AttackStart(Unit* /*victim*/) override
+    {
+    }
+
+    KarazhanChessOrientationType GetDefaultOrientationForTeam()
+    {
+        switch (me->GetFaction())
+        {
+            case CHESS_FACTION_ALLIANCE:
+                return ORI_NW;
+            case CHESS_FACTION_HORDE:
+                return ORI_SE;
+        }
+
+        return ORI_SE;
+    }
+
+    void CastChangeFacing(Creature* piece, Creature* trigger)
+    {
+        piece->CastSpell(trigger, SPELL_CHANGE_FACING, true);
+    }
+
+    void OnCharmed(bool apply) override
+    {
+        Unit* charmer = me->GetCharmer();
+        if (apply)
+        {
+            ASSERT(charmer);
+            _charmerGUID = charmer->GetGUID();
+            me->RemoveNpcFlag(UNIT_NPC_FLAG_GOSSIP);
+            me->SetWalk(true);
+            charmer->RemoveAurasDueToSpell(SPELL_GAME_IN_SESSION);
+
+            // Build new action bar
+            if (Player* playerCharmer = charmer->ToPlayer())
+            {
+                WorldPacket data(SMSG_PET_SPELLS, 1 + 8 + 2 + 4 + 1 + 1 + 2 + MAX_SPELL_CONTROL_BAR * 4 + 1 + 1);
+                data << me->GetGUID();
+                data << uint16(0);
+                data << uint32(0);
+                data << uint8(me->GetReactState());
+                data << uint8(0);
+                data << uint16(0);
+
+                for (uint32 i = 0; i < MAX_CREATURE_SPELLS; ++i)
+                {
+                    uint32 spellId = me->m_spells[i];
+                    SpellInfo const* spellInfo = sSpellMgr->GetSpellInfo(spellId);
+                    if (!spellInfo)
+                    {
+                        data << uint16(0) << uint8(0) << uint8(i + 8);
+                        continue;
+                    }
+
+                    data << uint32(MAKE_UNIT_ACTION_BUTTON(spellId, ACT_ENABLED));
+                }
+
+                for (uint32 i = MAX_CREATURE_SPELLS; i < MAX_SPELL_CONTROL_BAR; ++i)
+                    data << uint32(0);
+
+                data << uint8(0);
+                data << uint8(0);
+
+                playerCharmer->SendDirectMessage(&data);
+            }
+        }
+        else
+        {
+            me->SetNpcFlag(UNIT_NPC_FLAG_GOSSIP);
+
+            if (Unit* charmer = ObjectAccessor::GetUnit(*me, _charmerGUID))
+            {
+                charmer->RemoveAurasDueToSpell(SPELL_CONTROL_PIECE);
+                charmer->CastSpell(charmer, SPELL_GAME_IN_SESSION, true);
```

Comment:
```
if (instance->GetData(DATA_CHESS_EVENT) != DONE)
    charmer->CastSpell(charmer, SPELL_GAME_IN_SESSION, true);
```

Reply:
?? I don;t understand. Here player casts spell, not removes.
If the game is completed, why is he even trying to posses a piece? Even though, the aura be removed once leave the piece.

Reply:
I found in the test that when the chess is finished, it will return to the player again.

Reply:
1. "We want to play a game against you!"
2. Chess end
3. Players still have "SPELL_GAME_IN_ESSION"